File: TheBigDay/Models/Store.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using TheBigDay.Models.ItemCategoryModels;

namespace TheBigDay.Models
{
    [Table("Store")]
    public class Store
    {
        public Store() 
        {
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public int OperatingRadius { get; set; }
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? Suburb { get; set; }
        public string? State { get; set; }
        public string? Country { get; set; }
        public string? Postcode { get; set; }
        public string? ContactNum { get; set; }
        public string? AfterHoursContactName { get; set; }
        public string? AfterHoursContactNum { get; set; }
        public string? Email { get; set; }
        public string? PhotoPath { get; set; }
        public bool IsDeleted { get; set; }
        public bool IsActive { get; set; }
        public bool HasCompletedStoreSetup { get; set; }
        public StoreType? StoreType { get; set; }
        //store payment preferences
        public double? DepositPercentage { get; set; }
        public int? FullPaymentPrecedingEventDays { get; set; }


        #region navigation
        //public List<FormEntry>? Items { get; set; }
        public virtual List<Venue>? Venues { get; set; }
        //public List<Caterer> Caterers { get; set; }
        // public List<DiskJockey> Diskjockey { get; set; }
        public virtual List<User> Users { get; } = new List<User>();
        public virtual List<ItemCategory> ItemCategories { get; set; } = [];
        #endregion
    }
    public enum StoreType
    {
        INDIVIDUAL,
        BUSINESS,
    }
}

File: TheBigDay.Web/projects/common/src/lib/common-rest-models/store.ts
import {FormEntry} from "./form-entry";
import {ItemCategory} from "./item-category";

export interface Store {
    id: string;
    name: string;
    description: string;
    operatingRadius: number;
    addressLine1: string;
    addressLine2: string;
    suburb: string;
    state: string;
    country: string;
    postcode: string;
    contactNum: string;
    afterHoursContactNum: string;
    email: string;
    photoPath?: string;
    isDeleted: boolean;
    hasCompletedStoreSetup: boolean;
    storeType?: StoreType;
    items: FormEntry[];
    storeItemCategories: StoreItemCategory[];
    //store payment preference
    depositPercentage?: number;
    // if 0, then payment on the day. TODO: ask Zeal how this works? should thee payment be done before or after the event.
    fullPaymentPrecedingEventDays?: number;
}

export interface StoreItemCategory {
    id?: string,
    storeId: string,
    itemCategoryId: string,

    //nav
    itemCategory?: ItemCategory
    store?: Store
}


export enum StoreType {
    INDIVIDUAL,
    BUSINESS,
}
